File: frontend/src/types/empleado_profesional.ts
export interface FormacionProfesional {
  id_formacion?: number;
  nombre_titulo: string;
  ano: string;
  descripcion_especialidad: string;
}

export interface Idioma {
  id_idioma: number;
  nombre: string;
}

export interface EstructuraOrganizacional {
  id_museo: number;
  id_estructura_org: number;
  nombre: string;
  nivel: string;
  tipo: string;
  id_jerarquia_estructura?: number | null;
  id_jerarquia_museo?: number | null;
}

export interface HistoricoEmpleadoForm {
  id_museo: number;
  id_estructura_org: number;
  fecha_inicio: string;
  rol_empleado: string;
}

export interface EmpleadoProfesional {
  id_empleado_prof?: number;
  primer_nombre: string;
  segundo_nombre?: string;
  primer_apellido: string;
  segundo_apellido: string;
  fecha_nacimiento: string;
  doc_identidad: number;
  dato_contacto?: string;
  formaciones?: FormacionProfesional[];
  idiomas?: Idioma[];
}

export interface EmpleadoProfesionalFormData {
    primer_nombre: string;
    primer_apellido: string;
    segundo_apellido: string;
    fecha_nacimiento: string;
    doc_identidad: number;
    dato_contacto?: string;
    segundo_nombre?: string;
} 

File: schemas/empleado_profesional.py
from pydantic import BaseModel
from datetime import date
from typing import Optional, List
from .formacion_profesional import FormacionProfesionalCreate, FormacionProfesionalResponse
from .idioma import IdiomaResponse

class EstructuraOrganizacionalResponse(BaseModel):
    id_museo: int
    id_estructura_org: int
    nombre: str
    nivel: str
    tipo: str
    id_jerarquia_estructura: Optional[int] = None
    id_jerarquia_museo: Optional[int] = None

    class Config:
        from_attributes = True

class HistoricoEmpleadoCreate(BaseModel):
    id_museo: int
    id_estructura_org: int
    fecha_inicio: date
    rol_empleado: str

class EmpleadoProfesionalBase(BaseModel):
    primer_nombre: str
    primer_apellido: str
    segundo_apellido: str
    fecha_nacimiento: date
    doc_identidad: int
    dato_contacto: Optional[str] = None
    segundo_nombre: Optional[str] = None

class EmpleadoProfesionalCreate(EmpleadoProfesionalBase):
    formaciones: Optional[List[FormacionProfesionalCreate]] = None
    idiomas: Optional[List[int]] = None  # Lista de IDs de idiomas
    historico: HistoricoEmpleadoCreate

class EmpleadoProfesionalResponse(EmpleadoProfesionalBase):
    id_empleado_prof: int
    formaciones: List[FormacionProfesionalResponse] = []
    idiomas: List[IdiomaResponse] = []

    class Config:
        from_attributes = True 